v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/WriteToGCSParquet.java

```diff
@@ -0,0 +1,144 @@
+
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.io.FileIO;
+import org.apache.beam.sdk.io.WriteFilesResult;
+import org.apache.beam.sdk.io.parquet.ParquetIO;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WriteToGCSParquet} class is a {@link PTransform} that takes in {@link PCollection} of
+ * KV records.The Transform converts and writes these records to GCS in Parquet File Format.
+ */
+@AutoValue
+public abstract class WriteToGCSParquet
+    extends PTransform<PCollection<KV<String, String>>, WriteFilesResult<Void>> {
+
+  /* Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(WriteToGCSText.class);
+
+  @VisibleForTesting protected static final String DEFAULT_OUTPUT_FILE_PREFIX = "output";
+
+  /**
+   * The {@link WriteToGCSParquetOptions} interface provides the custom execution options passed by
+   * the executor at the command-line.
+   */
+  public interface WriteToGCSParquetOptions extends PipelineOptions {
+
+    @Description("The directory to output files to. Must end with a slash. ")
+    String getOutputDirectory();
+
+    void setOutputDirectory(String outputDirectory);
+
+    @Description(
+        "The filename prefix of the files to write to. Default file prefix is set to \"output\". ")
+    String getOutputFilenamePrefix();
+
+    void setOutputFilenamePrefix(String outputFilenamePrefix);
+
+    @Description(
+        "The maximum number of output shards produced when writing. Default number is runner defined. ")
+    @Default.Integer(0)
+    Integer getNumShards();
+
+    void setNumShards(Integer numShards);
+  }
+
+  public static WriteToGCSBuilder newBuilder() {
+    return new AutoValue_WriteToGCSParquet.Builder();
+  }
+
+  public abstract String outputDirectory();
+
+  public abstract String outputFilenamePrefix();
+
+  public abstract Integer numShards();
+
+  @Override
+  public WriteFilesResult<Void> expand(PCollection<KV<String, String>> kafkaRecords) {
+    return kafkaRecords
+        /*
+         * Converting KV<String, String> records to GenericRecord using DoFn and {@link ConstructGenericRecordsFn} class.
+         */
+        .apply("Create GenericRecord(s)", ParDo.of(new ConstructGenericRecordsFn()))
+        .setCoder(AvroCoder.of(GenericRecord.class, WriteToGCSUtility.SCHEMA))
+        /*
+         * Writing as parquet file using {@link FileIO} and {@link ParquetIO}.
+         *
```

Comment:
The formatting looks off. Can you use the [google-java-format Intellij plugin](https://plugins.jetbrains.com/plugin/8527-google-java-format) to help automate some of this?